routes/create-or-update-file.js

```diff
@@ -0,0 +1,65 @@
+var request = require("request");
+
+module.exports = function(config) {
+  return function(req, res) {
+    if(!config.authorized(req, res)) {
+      return;
+    }
+
+    if(!req.body || !req.body.path || !req.body.buffer) {
+      // TODO: handle error
+      console.error('Request body missing data: ', req.body);
+      res.send(400);
+      return;
+    }
+
+    var fileReceived = {
+      path: req.body.path,
+      buffer: req.body.buffer.data,
+      project_id: req.session.project.meta.id
+    };
+    var existingFile = req.session.project.files[fileReceived.path];
```

Comment:
Yeah that's true! I'm not a huge fan of storing file lists on the server, but for now I don't think there is a clean way around it. Luckily, we're not storing the buffers.